presto-parser/src/main/java/com/facebook/presto/sql/parser/SqlParser.java

```diff
@@ -108,6 +114,13 @@ private Node invokeParser(String name, String sql, Function<SqlBaseParser, Parse
         }
     }
 
+    private static synchronized void clearDFA(SqlBaseParser parser)
+    {
+        if (++invocationCount % 1000 == 0) { // is it too often?
```

Comment:
I am just curious to know how this is better/safer than below.
~~~
if (++invocationCount >= 1000) {
    invocationCount = 0;
    ...
}
~~~